### PR TITLE
Upgrade CARTS to AWS SDK v3, mostly

### PR DESCRIPTION
**Pull request by benmartin-coforma**
### Description
It's happening! CARTS is moving to the new SDK version. Here's the service-by-service breakdown:
* root: ❌ - not upgraded. We don't use SDKv2 directly, but a few of our serverless plugins do (serverless-cloudfront-invalidator, serverless-s3-sync, and serverless itself). There's not much we can do about this; hopefully these packages move to v3 in a timely manner.
* app-api: ✅ - upgraded. I replaced all usage of v2 for S3, DynamoDB, and SSM. We still have a dependency by way of aws-lambda, but I was able to make that a "soft" dependency by copying the one type definition we currently need.
* carts-bigmac-streams: ❌ - not upgraded. Any changes I made here would conflict with existing work in progress related to the BigMac IAM effort. We will want to coordinate the v3 change and any testing with the BigMac folks. I will create a new story in the backlog to track this.
* database: ✅ - upgraded. I replaced the usage of v2 for DynamoDB in the local database seed function.
* ui: ✅ - no change needed.
* ui-auth: ✅ - upgraded. I replaced the usage of v2 for Cognito in the test user seed function.
* ui-src: ✅ - no change needed. The only edits in here are to stay compatible with app-api.
* uploads: ❌ - not upgraded. The virus-scanning code involves streaming data from S3, and as far as I can tell the v3 S3 client requires nodeJS version 17 or higher for this. CARTS is currently running node 16. It's possible there is a workaround I wasn't able to find, and it's possible that our best path forward is to upgrade node. I will create a new story in the backlog to track this.

_EDIT_: I have removed two more v2 dependencies from app-api:
* Previously we injected loggers into all AWS clients by way of the global `AWS.config` object (a v2 construct). Now we have a reworked logger with a simpler interface, we log more of each request (see handler-lib), and the logger is provided manually to all three AWS clients (SSM, S3, and DynamoDB).
* Previously we signed requests to the Prince PDF API using credentials extracted from the global `AWS.config` object. Now we are pulling those credentials out of `process.env` instead, where as far as I can tell they were magically injected by AWS.
   * While I was in there, I replaced the `aws4` signature package (don't be confused: that's version 4 of the AWS signature algorithm, not the SDK) with a package from `@smithy` (which I believe is used within AWS for code generation, so it's as official as we can get). I also replaced `axios` with `fetch`, which is built into Node 18+.
   * You'll notice that `fetch` is currently imported from `cross-fetch`; this is a polyfill which is only necessary because CARTS is still on Node 16. As soon as we upgrade the node version, we can uninstall `cross-fetch` with no futher code changes.

### Related ticket(s)
CMDCT-2689

---
### How to test
1. app-api: SSM
   * Ensure you can log in to the app.
2. app-api: Dynamo
   * Ensure forms can be generated - this tests the scanAll and batchWriteItem functions of dynamodb-lib
   * Ensure you can enter a report and change a value. Then leave and come back. This tests the query and update functions.
3. app-api: S3
   * On the state user home screen there is a button to download the fiscal year template. Ensure it works.
   * In Section 2, the very last question has a file upload. Ensure you can upload a file.
   * Then wait a minute (to give the AV scan time to run) and ensure you can download that file.
   * Then delete the file. This tests the delete function dynamodb-lib: the fact that a file has been attached is stored in the database.
   * Finally, verify in AWS that the file was removed from the bucket. This tests the delete function of s3-lib.
4. app-api: debug-lib
   * Ensure lambdas are writing their activity to CloudWatch. The logging configuration is one of two remaining references to `aws-sdk` in app-api
5. app-api: prince
   * Ensure that you can print a report to PDF. The code to authenticate with prince also references `aws-sdk`.
6. database: DynamoDB
   * Note that this branch's deployed environment had some pre-seeded state reports.
7. ui-auth: Cognito
   * Note that this branch has a populated Cognito identity pool in AWS. (We don't _use_ this pool; all feature branches point to the `main` identity pool when authenticating users. But they each populate their own pool anyway, with this code).
8. ui-src: You already tested the changes here, in order to test S3.
9. Search the entire repo for this regular expression: `(?<!@)aws-sdk(?!-client-mock)`. You'll see that every instance of `aws-sdk` (which is neither preceded by `@` nor followed by `-client-mock`) is accounted for in this PR description.
   * Except for an ETL which was written after I started work on this, and I'm not going to modify an in-flight ETL. We'll delete it once it's run in all environments.
   * _Edit_: now fewer instances than ever, thanks to the work on debug-lib.ts and printPdf.ts!

### Important updates
You may have to run `yarn`?

And please don't `yarn add aws-sdk` anywhere; that's the v2 package.

---
### Author checklist

- [x] I have performed a self-review of my code
- [x] I have added [thorough](https://shorturl.at/aejkF) tests, if necessary
- ~[ ] I have updated relevant documentation, if necessary~

### CODE REVIEW

**Review by gmrabian — DISMISSED** [review]
Reviewed code, ran locally, tested deployed url. Everything here checks out 😎 
Outstanding stuff, Ben!

**Review by karla-vm — DISMISSED** [review]
this is huge, Ben! 🎉 awesome work.

File: services/app-api/libs/debug-lib.ts [hunk 1]
```diff
@@ -2,7 +2,7 @@
 /* eslint-disable no-console */
 import util from "util";
 import AWS from "aws-sdk";
-import { APIGatewayProxyEvent } from "aws-lambda"; // eslint-disable-line no-unused-vars
+import { APIGatewayProxyEvent } from "../types"; // eslint-disable-line no-unused-vars
```
[the remark this answers is not in the document]
Comment (benmartin-coforma) [reply services/app-api/libs/debug-lib.ts:5]: Gah! I thought I caught all these >_<

**Comment by benmartin-coforma** [comment]
The spin-off stories for carts-bigmac-streams and uploads are CMDCT-3078 and CMDCT-3079

**Comment by benmartin-coforma** [comment]
This blog post was really helpful in getting the signature code up & working: https://dev.to/aws-builders/signing-requests-with-aws-sdk-in-lambda-functions-476

You'll notice he's using the package `@aws-sdk/signature-v4`, which [was moved](https://www.npmjs.com/package/@aws-sdk/signature-v4) to the `@smithy` NPM organization, about 5 months ago it seems.

**Review by karla-vm — APPROVED** [review]

**Review by britt-mo — APPROVED** [review]
Thanks for the walkthrough yesterday!

**Pull request merged**